text/1071-cell.md

```diff
@@ -0,0 +1,296 @@
+
+    /**
+     * Prevents further updates, making the Cell
+     * behave as a ReadOnlyReactive
+     */
+    freeze: () => void;
+}
+```
+
+Behaviorally, the `Cell` behaves almost the same as this function:
+```js
+class CellPolyfill {
+    @tracked current;
+    #isFrozen = false;
+
+    constructor(initialValue) {
+        this.current = initialValue;
+    }
+
+    read() {
+        return this.current;
+    }
+
+    set(value) {
+        assert(`Cannot set a frozen Cell`, !this.#isFrozen);
+        this.current = value;
+    }
+
+    update(updater) {
+        assert(`Cannot update a frozen Cell`, !this.#isFrozen);
+        this.set(updater(this.read()));
+    }
+
+    freeze() {
+        this.#isFrozen = true;
+    }
+}
+```
+
+The key difference is that with a primitive, we expose a new way for developers to decide when their value becomes dirty.
+The above example, and the default value, would use the "always dirty" behavior of `() => false`.
+
+This default value allows the `Cell` to be the backing implementation if `@tracked`, as `@tracked` values do not have equalty checking to decide when to become dirty.
+
+For example, with this Cell and equality function:
+
+```gjs
+const value = Cell.create(0, (a, b) => a === b);
+
+const selfAssign = () => value.current = value.current;
+
+<template>
+    <output>{{value}}</output>
+
+    <button {{on 'click' selfAssign}}>Click me</button> 
+</template>
+```
+
+The contents of the `output` element would never re-render due to the value never changing. 
+
+This differs from `@tracked`, as the contents of `output` would always re-render.
+
+
+### Usage
+
+Incrementing a count with local state.
+
+```gjs
+import { Cell } from '@glimmer/tracking';
+
+const increment = (cell) => cell.current++;
+
+<template>
+    {{#let (Cell.create @initialCount) as |count|}}
+       Count is: {{count.current}} 
+
+        <button {{on "click" (fn increment count)}}>add one</button>
+    {{/let}}
+</template>
+```
+
+Incrementing a count with module state.
+This is already common in demos.
+
+```gjs
+import { Cell } from '@glimmer/tracking';
+
+const count = Cell.create(0);
+const increment => count.current++;
+
+<template>
+   Count is: {{count.current}} 
+
+    <button {{on "click" increment}}>add one</button>
+</template>
+```
+
+Using private mutable properties providing public read-only access:
+
+```gjs
+export class MyAPI {
+    #state = Cell.create(0);
+
+    get myValue() {
+        return this.#state;
+    }
+
+    doTheThing() {
+        this.#state = secretFunctionFromSomewhere(); 
+    }
+}
+```
+
+
+### Re-implementing `@tracked` 
+
+For most current ember projects, using the TC39 Stage 1 implementation of decorators:
+
+```js
+function tracked(target, key, { initializer }) {
+  let cells = new WeakMap();
+
+  function getCell(obj) {
+    let cell = cells.get(obj);
+
+    if (cell === undefined) {
+      cell = Cell.create(initializer.call(this), () => false);
```

Comment:
TODO: `Cell`s need a debug-time description so that we have better autotracked-debugging.

This can be stored in a WeakMap or something so that other tools can access the name.

This is important because Cells are not always going to be on class properties